File: backend-dotnet/Controllers/VehicleDetailController.cs
using backend_dotnet.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace backend_dotnet.Controllers
{
    [ApiController]
    [Route("api/default-config")]
    [Authorize]
    public class VehicleDetailController : ControllerBase
    {
        private readonly IVehicleDetailService _vehicleDetailService;

        public VehicleDetailController(
            IVehicleDetailService vehicleDetailService)
        {
            _vehicleDetailService = vehicleDetailService;
        }

        [HttpGet("{modelId}")]
        public async Task<IActionResult> GetDefaultConfiguration(
            int modelId)
        {
            var response =
                await _vehicleDetailService
                    .GetDefaultConfigurationAsync(modelId);

            if (response == null)
            {
                return NotFound();
            }

            return Ok(response);
        }
    }
}

File: backend/java/Vehicle_Configurator/src/main/java/com/example/demo/controllers/VehicleDetailController.java
package com.example.demo.controllers;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;


import com.example.demo.dto.DefaultConfigurationDTO;
import com.example.demo.services.VehicleDetailService;



@RestController
@RequestMapping("/api/default-config")
@CrossOrigin("*")
public class VehicleDetailController {



    private VehicleDetailService service;



    public VehicleDetailController(
            VehicleDetailService service){

        this.service = service;

    }




    @GetMapping("/{modelId}")
    public ResponseEntity<?> getDefaultConfiguration(
            @PathVariable int modelId){



        DefaultConfigurationDTO response =
                service.getDefaultConfiguration(modelId);



        if(response==null){

            return ResponseEntity
                    .notFound()
                    .build();

        }



        return ResponseEntity.ok(response);

    }


}